library/core/src/slice/iter.rs

```diff
@@ -184,6 +189,11 @@ unsafe impl<T: Sync> Sync for IterMut<'_, T> {}
 unsafe impl<T: Send> Send for IterMut<'_, T> {}
 
 impl<'a, T> IterMut<'a, T> {
+    pub(super) unsafe fn new(ptr: *mut T, end: *mut T) -> Self {
+        // SAFETY: pointers passed to this function must be valid.
+        Self { ptr: unsafe { NonNull::new_unchecked(ptr) }, end, _marker: PhantomData }
```

Comment:
would it make sense to instead pass a `NonNull` here for `ptr`

Reply:
I wanted to move the unsafe into `new`. Looks like we don't need it cause it used already in unsafe block.